### PR TITLE
Fix CI errors

### PR DESCRIPTION
**Pull request by cpjordan**
Fix various issues with testing:

- ufl_element() -> ufl_function_space()
- Fatal Python error: Aborted - TO BE FIXED
- AttributeError: 'Cofunction' object has no attribute '_ufl_class_' - TO BE FIXED

### CODE REVIEW

File: thetis/callback.py [hunk 1]
```diff
@@ -996,7 +996,7 @@ def __init__(self, solver_obj, fieldnames, x, y,
         self.field_dims = {}
         for f in self.fieldnames:
             func = solver_obj.fields[f]
-            self.field_dims[f] = func.function_space().value_size
+            self.field_dims[f] = func.function_space().block_size
```
Comment (stephankramer) [inline thetis/callback.py:999]: Same here

File: thetis/exporter.py [hunk 2]
```diff
@@ -22,11 +22,11 @@ def get_visu_space(fs):
     """
     mesh = fs.mesh()
     family = 'Lagrange' if is_cg(fs) else 'Discontinuous Lagrange'
-    if len(fs.ufl_element().value_shape) == 1:
-        dim = fs.ufl_element().value_shape[0]
+    if len(fs.ufl_function_space().value_shape) == 1:
```
Comment (stephankramer) [inline thetis/exporter.py:25]: Does `fs.value_shape` not work?

File: thetis/callback.py [hunk 3]
```diff
@@ -508,7 +508,7 @@ def __init__(self, solver_obj,
         """
         # printing all detector output to log is probably not a useful default:
         kwargs.setdefault('append_to_log', False)
-        self.field_dims = [solver_obj.fields[field_name].function_space().value_size
+        self.field_dims = [solver_obj.fields[field_name].function_space().block_size
```
Comment (stephankramer) [inline thetis/callback.py:511]: I'm a little confused tbh. I would have expected this should be `value_size` as that would correspond to the size of what you get when interpolating in a point (i.e. the size of the value) - which is not necessarily the same as the n/o DOFs per node which is what `block_size` gives IIUC. I know that your change should give back the old behaviour. Might it be that `DetectorsCallback` just has not been tested with RT or BDFM velocity fields?

File: examples/discrete_turbines/turbine_callback.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def __init__(self, solver_obj,
         self.field_dims = []
         for field_name in field_names:
             if field_name != 'pow':
-                self.field_dims.append(solver_obj.fields[field_name].function_space().value_size)
+                self.field_dims.append(solver_obj.fields[field_name].function_space().block_size)
```
Comment (stephankramer) [inline examples/discrete_turbines/turbine_callback.py:35]: See comment for DetectorsCallback

**Review by stephankramer — COMMENTED** [review]
Thanks @cpjordan 
So it might be I'm just misunderstanding - but I would have expected that for interpolated value size we should in fact be using `value_size` and not `block_size`. The latter being the n/o DOFs per node, which was what previously returned by `value_size` in which case it would in fact have been incorrect in Thetis previously. The difference would only appear for RT and BDM velocity fields which we don't use very frequently. I do agree/understand the changes to the OP2 kernel headers in utility.py and utility3d.py - these should indeed be `block_size` now.

**Comment by stephankramer** [comment]
Related to https://github.com/firedrakeproject/firedrake/pull/3862 and https://github.com/firedrakeproject/firedrake/pull/3362

**Comment by cpjordan** [comment]
@stephankramer - thanks for reviewing and explanation. I had discussed the problems with @connorjward who found that `value_size` in the utility kernels was the cause of the failing tests, but I just replaced all instances assuming that the previous implementation in Thetis was correct - I should have looked into it more!

Updated based on the comments above.

**Review by stephankramer — APPROVED** [review]
Perfect, many thanks!

**Pull request merged**